src/cloudinary-placeholder.component.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  AfterContentChecked,
+  Component,
+  HostBinding,
+  Input,
+} from '@angular/core';
+import {Cloudinary} from './cloudinary.service';
+
+
+@Component({
+  selector: 'cl-placeholder',
+  template: `<img [src]="this.smallImg" [style.width.px]="this.itemWidth"[style.height.px]="this.itemHeight">`
+  ,
+})
+export class CloudinaryPlaceHolder implements AfterContentChecked {
+  @Input('type') type: string;
+  @HostBinding('style.width') itemWidth;
+  @HostBinding('style.height') itemHeight;
+  @HostBinding('attr.public-id') publicId;
+
+  options: object = {};
+  smallImg: string;
+
+  constructor(private cloudinary: Cloudinary) {}
+
+  setWidth(width) {
+    this.itemWidth = width;
+  }
+
+  setHeight(height) {
+    this.itemHeight = height;
+  }
+
+  setPublicId(id) {
+    this.publicId = id;
+  }
+
+  ngAfterContentChecked() {
+    this.smallImg = this.getSmallImage();
+  }
+
+  getSmallImage() {
```

Comment:
@strausr , please rename to getPlaceholderImage

Reply:
done